### PR TITLE
Filter request from url

### PR DESCRIPTION
**Pull request by NicolasAubry**

### CODE REVIEW

File: hammertime/rules/filterrequestfromurl.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+    def _match_found(self, url, filter_list):
+        parsed = urlparse(url)
+        for filter in filter_list:
+            if self._apply_filter(parsed, filter):
+                return True
+        return False
+
+    def _apply_filter(self, parsed_url, filter):
+        for key in filter.keys():
```
Comment (lphuberdeau) [inline hammertime/rules/filterrequestfromurl.py]: for key, value in  filter.items():

File: tests/rules/filterrequestfromurl_test.py [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+    def setUp(self):
+        pass
+
+    @async_test()
+    async def test_before_request_allow_request_to_host_matching_filter_in_whitelist(self):
+        filter = FilterRequestFromURL(allowed_urls="example.com")
+
+        try:
+            await filter.before_request(Entry.create("http://example.com/"))
+        except RejectRequest as e:
```
Comment (lphuberdeau) [inline tests/rules/filterrequestfromurl_test.py]: Not necessary, thrown exceptions cause the test to fail anyway.

File: tests/rules/filterrequestfromurl_test.py [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+            await filter.before_request(Entry.create("http://example.com/allowed/index.html"))
+        except RejectRequest as e:
+            self.fail(str(e))
+
+    @async_test()
+    async def test_before_request_reject_request_to_path_not_matching_filter_in_whitelist(self):
+        filter = FilterRequestFromURL(allowed_urls="/allowed")
+
+        with self.assertRaises(RejectRequest):
+            await filter.before_request(Entry.create("http://example.com/test/index.html"))
```
Comment (lphuberdeau) [inline tests/rules/filterrequestfromurl_test.py:68]: What about http://example.com/allowed-oops ?

**Review by lphuberdeau — CHANGES_REQUESTED** [review]

File: tests/rules/filterrequestfromurl_test.py (hunk 3 above)
Comment (NicolasAubry) [reply tests/rules/filterrequestfromurl_test.py:68]: Now fixed and tested by https://github.com/delvelabs/hammertime/pull/25/files#diff-bd6cbceded6380d894ad8478634e1dccR138

**Review by lphuberdeau — APPROVED** [review]

**Pull request merged**